### DISCUSSION TITLE
Redefining name 'app' from outer scope when using lifespan()

### DISCUSSION THREAD

**Discussion by diamond-fish** [fastapi/fastapi#10543] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from contextlib import asynccontextmanager

from fastapi import FastAPI


def fake_answer_to_everything_ml_model(x: float):
    return x * 42


ml_models = {}


@asynccontextmanager
async def lifespan(app: FastAPI):
    # Load the ML model
    ml_models["answer_to_everything"] = fake_answer_to_everything_ml_model
    yield
    # Clean up the ML models and release the resources
    ml_models.clear()


app = FastAPI(lifespan=lifespan)


@app.get("/predict")
async def predict(x: float):
    result = ml_models["answer_to_everything"](x)
    return {"result": result}
```

### Description

When using the sample code on https://fastapi.tiangolo.com/advanced/events/, pylint is throwing the "Redefining name 'app' from outer scope" warning. Is this something that should be expected and we should ignore the warning?

### Operating System

Windows

### FastAPI Version

0.103.2

### Pydantic Version

2.4.2

### Python Version

3.12.0

**Comment by NiKuma0** [comment, marked as answer]
You shouldn't blindly follow every rule that Pylint (or any other linter) provides. Just think about what problem it would cause if you redefined it. If none, then just turn off that annoying warning for that line or for the whole project. If you think there might be a problem, then just rename it (e.g. `_app` or `app_`).

In this case my opinion is that renaming is good to go (simply because disabling will take more time). But if you make big project, just make new file where you put all your lifespans, so there would be no such problem.
